### PR TITLE
feat: url validation

### PR DESCRIPTION
**Pull request by braianj**
###  Summary

- Adds secure URL validation utilities (validateUrl, validateUrlInstance)
- Includes checks for protocols, ports, characters, and SSRF protection
- Adds comprehensive test coverage for multiple edge cases
- Updates public API and type definitions accordingly

### CODE REVIEW

File: src/core/url-validation.ts [hunk 1]
```diff
@@ -0,0 +1,652 @@
...
+  export const validate: ValidateFunction<string> = generateLazyValidator(schema, [_isSafeUrl])
+
+  /**
+   * Safely builds a URL with query parameters, automatically encoding all values
+   * @param basePath - The base path without query parameters
+   * @param params - Object containing query parameters
+   * @returns A properly encoded URL string
+   * @throws Error if malicious content is detected
+   */
+  export function buildUrlWithParams(basePath: string, params: QueryParams): string {
```
Comment (LautaroPetaccio) [inline src/core/url-validation.ts:292]: What if, instead of having an URL builder, which the URL class already has (and does it correctly), we have a URL instance checker? We can even convert the URL to string and use the string checker that was defined before.

File: src/core/url-validation.ts [hunk 2]
```diff
@@ -0,0 +1,652 @@
...
+
+    return parsed.url.searchParams.get(key)
+  }
+
+  /**
+   * Validates if a complete URL is safe
+   * @param url - The URL to validate
+   * @returns true if the URL is safe, false if it contains malicious content
+   */
+  export function isSafeUrl(url: string): boolean {
```
Comment (LautaroPetaccio) [inline src/core/url-validation.ts:464]: Isn't these checks the same / similar than the on in `_isSafeUrl`?
